lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -59,10 +62,16 @@ def build_index(self, server_config, dates):
             dates (iterable): list of date strings
         """
         idx = server_config.get("Indexing", "index_prefix") + ".v6.run-data."
-        index = "/"
-        for d in dates:
-            index += f"{idx}{d},"
-        return index
+        index = "".join([f"{idx}{d}," for d in dates])
+
+        return f"/{index}"
+
+    def build_index_from_metadata(self):
+        drb = Dataset.attach(controller="node", name="drb")
+        index_map = Metadata.getvalue(dataset=drb, key="server.index-map")
+        index_keys = [key for key in index_map if "result-data-sample" in key]
+        assert len(index_keys) == 1
```

Comment:
Different problem than in the production server code, but this `assert` is also ... awkward ... since it means that the mocked document index is set up incorrectly. If that's a case we want to cover in a unit test, we'd be looking for the API to fail with an internal server error (not an `assert`) and otherwise this isn't going to be really helpful to someone who hits it in testing. This might be a case where the optional assert message would help: this means that the document index map has a duplicate root index name!